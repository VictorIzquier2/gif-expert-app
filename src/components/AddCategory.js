import React, {Fragment, useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value)

    console.log('handleInputChange llamado');
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    console.log('handleSubmit', inputValue);

    if(inputValue.trim().length > 0){
      setCategories(cats => [inputValue, ...cats])
      setInputValue('');
    }
  }

  return (
    <Fragment>
      <form
        onSubmit={handleSubmit}
      >
        <p>{inputValue}</p>
        <input
          type='text'
          value={inputValue}
          onChange={handleInputChange}
        />
      </form>
    </Fragment>
   );
}
AddCategory.protoTypes = {
  setCategories: PropTypes.func.isRequired
}
 
export default AddCategory;