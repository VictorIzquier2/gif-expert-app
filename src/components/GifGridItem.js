import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({url, title}) => {
  return ( 
    <Fragment>
      <div className='card'>
        <img
          src={url}
          alt={title}
          height={240}
        />
        <p>{title}</p>
      </div>
    </Fragment>
   );
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
export default GifGridItem;
