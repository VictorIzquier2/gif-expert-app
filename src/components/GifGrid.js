import { Fragment } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

  const {data: images, loading}= useFetchGifs(category)

  return ( 
    <Fragment>
      <h3>{category}</h3>
      {loading && 'Cargando...'}
      <div className='card-grid'>
        {images.map(img => (
          <GifGridItem
            key={img.id}
            {...img}
          />
        ))}
      </div>

    </Fragment>
   );
}
GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifGrid;