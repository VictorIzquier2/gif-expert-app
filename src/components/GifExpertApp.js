import React, { Fragment, useState } from 'react';
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = () => {

  //const categories = ['One Punch', 'Samurai X', 'Dragon Ball']
  const [categories, setCategories] = useState(
    ['']
  )

  return (
    <Fragment>
      <h2>GifExpertApp</h2>
      <AddCategory
        setCategories={setCategories}
      />
      <hr/>

      <ol>
        {categories.map(category => (
          <GifGrid 
            category={category}
            key={category}
          />
        ))}
      </ol>
    </Fragment>
   );
}
 
export default GifExpertApp;