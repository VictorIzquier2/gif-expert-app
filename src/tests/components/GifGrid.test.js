import React from 'react';
import '@testing-library/jest-dom'
import {shallow} from 'enzyme';
import {GifGrid} from '../../components/GifGrid';
import {useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el <GifGrid />', () => {

  const category = 'One Punch';
  
  test('Debe de mostrarse correctamente', () => {
    
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();

  });

  test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

    const gifs = [
      {
        id: 'Dragon Ball',
        url: 'https://media3.giphy.com/media/DzeKll1HYxiYo/giphy.gif?cid=f18a8f9bn8frrg5r778mjk79cowvlr8ixxmqp121maxk18bz&rid=giphy.gif',
        title: 'Dragon Ball'
      },
      {
        id: 'Dragon Ball',
        url: 'https://media3.giphy.com/media/DzeKll1HYxiYo/giphy.gif?cid=f18a8f9bn8frrg5r778mjk79cowvlr8ixxmqp121maxk18bz&rid=giphy.gif',
        title: 'Dragon Ball'
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

  });

});