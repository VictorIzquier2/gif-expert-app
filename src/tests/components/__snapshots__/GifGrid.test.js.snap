// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  Cargando...
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Dragon Ball"
      key="Dragon Ball"
      title="Dragon Ball"
      url="https://media3.giphy.com/media/DzeKll1HYxiYo/giphy.gif?cid=f18a8f9bn8frrg5r778mjk79cowvlr8ixxmqp121maxk18bz&rid=giphy.gif"
    />
    <GifGridItem
      id="Dragon Ball"
      key="Dragon Ball"
      title="Dragon Ball"
      url="https://media3.giphy.com/media/DzeKll1HYxiYo/giphy.gif?cid=f18a8f9bn8frrg5r778mjk79cowvlr8ixxmqp121maxk18bz&rid=giphy.gif"
    />
  </div>
</Fragment>
`;
